cmd/git-sync/main.go

```diff
@@ -763,7 +778,13 @@ func main() {
 
 		log.V(1).Info("next sync", "waitTime", flPeriod.String())
 		cancel()
-		time.Sleep(*flPeriod)
+
+		t := time.NewTimer(*flPeriod)
+		select {
+		case <-t.C:
+		case <-signal_ch:
+			t.Stop()
```

Comment:
docs for Timer.Stop say:

```
   To ensure the channel is empty after a call to Stop, check the return value
    and drain the channel. For example, assuming the program has not received
    from t.C already:

        if !t.Stop() {
        	<-t.C
        }
```

Reply:
Yes, that documentation mentions the necessity of draining the channel before calling Reset(). Do we need to drain the channel otherwise? Or does GC take care of that?

Reply:
I just realized that this is scoped to the loop, so we shouldn't need to worry about it.